test/utils/Arrays.test.js

```diff
@@ -1,88 +1,104 @@
-require('@openzeppelin/test-helpers');
-
+const { ethers } = require('hardhat');
 const { expect } = require('chai');
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
+const { randomHex } = require('../helpers/random');
 
-const AddressArraysMock = artifacts.require('AddressArraysMock');
-const Bytes32ArraysMock = artifacts.require('Bytes32ArraysMock');
-const Uint256ArraysMock = artifacts.require('Uint256ArraysMock');
+const AddressArraysMock = 'AddressArraysMock';
+const Bytes32ArraysMock = 'Bytes32ArraysMock';
+const Uint256ArraysMock = 'Uint256ArraysMock';
 
-contract('Arrays', function () {
+describe('Arrays', function () {
   describe('findUpperBound', function () {
-    context('Even number of elements', function () {
+    describe('Even number of elements', function () {
       const EVEN_ELEMENTS_ARRAY = [11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
 
+      const fixture = async () => {
+        const arrays = await ethers.deployContract(Uint256ArraysMock, [EVEN_ELEMENTS_ARRAY]);
+        return { arrays };
+      };
+
       beforeEach(async function () {
-        this.arrays = await Uint256ArraysMock.new(EVEN_ELEMENTS_ARRAY);
+        Object.assign(this, await loadFixture(fixture));
       });
 
       it('returns correct index for the basic case', async function () {
-        expect(await this.arrays.findUpperBound(16)).to.be.bignumber.equal('5');
+        expect(await this.arrays.findUpperBound(16)).to.be.equal('5');
```

Comment:
We should use bigint everywhere numbers are involved.
```suggestion
        expect(await this.arrays.findUpperBound(16)).to.be.equal(5n);
```